specification.md

```diff
@@ -5026,7 +5071,124 @@ for your workflow definitions especially in production environments.
+
+Globals can be defined via the [Workflow top-level "globals" property](#Workflow-Definition-Structure),
+for example:
+
+```json
+"globals": {
+  "Translations": {
+    "Dog": {
+      "Serbian": "pas",
+      "Spanish": "perro",
+      "French": "chien"
+    }
+  }
+}
+```
+
+Globals can only be accessed inside Workflow expressions via the $GLOBALS namespace.
+Runtimes must make globals available to expressions under that namespace.
+
+Here is an example of using globals in Workflow expressions:
+
+```json
+{
+...,
+"globals": {
+  "AGE": {
+    "MIN_ADULT": 18
+  }
+},
+...
+"states":[  
+  {  
+     "name":"CheckApplicant",
+     "type":"switch",
+     "dataConditions": [
+        {
+          "name": "Applicant is adult",
+          "condition": "${ .applicant | .age >= $GLOBALS.AGE.MIN_ADULT }",
+          "transition": "ApproveApplication"
+        },
+        {
+          "name": "Applicant is minor",
+          "condition": "${ .applicant | .age < $GLOBALS.AGE.MIN_ADULT }",
+          "transition": "RejectApplication"
+        }
+     ],
+     ...
+  },
+  ...
+]
+}
+```
+Note that globals can also be used in [expression functions](#Using-Functions-for-Expression-Evaluation),
+for example:
+
+```json
+{
+"functions": [
+  {
+    "name": "isAdult",
+    "operation": ".applicant | .age >= $GLOBALS.AGE.MIN_ADULT",
+    "type": "expression"
+  },
+  {
+    "name": "isMinor",
+    "operation": ".applicant | .age < $GLOBALS.AGE.MIN_ADULT",
+    "type": "expression"
+  }
+]
+}
+```
+
+Workflow globals values should only contain static data, meaning that their value should not 
+contain Workflow expressions. 
+Workflow globals should not have access to [Workflow secrets definitions](#Workflow-Secrets).
+
+### Workflow Secrets
+
+Secrets allow you access sensitive information, such as passwords, OAuth tokens, ssh keys, etc
+inside your [Workflow Expressions](#Workflow-Expressions). 
+
+You can define the names of secrets via the [Workflow top-level "secrets" property](#Workflow-Definition-Structure), 
+for example:
+
+```json
+"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+```
+
+If secrets are defined in a Workflow definition, runtimes must assure to provide their values
+during Workflow execution.
+
+Secrets can be used only in [Workflow expressions](#Workflow-Expressions) under the `SECRETS` namespace. 
+This is reserved namespace that should only be allowed for values defined by the `secrets` property.
+
+Here is an example on how to use secrets and pass them as arguments to a function invocation:
+
+```json
+"secrets": ["AZURE_STORAGE_ACCOUNT", "AZURE_STORAGE_KEY"],
+
+...
+        
+{
+  "refName": "uploadToAzure",
+    "arguments": {
+      "account": "${ $SECRETS.AZURE_STORAGE_ACCOUNT }",
+      "account-key": "${ $SECRETS.AZURE_STORAGE_KEY }",
+      ...
+    }
+  
+}
+```
+
+Note that secrets can also be used in [expression functions](#Using-Functions-for-Expression-Evaluation).
+
```

Comment:
Are secrets mutable, i.e. can I use an action to obtain a secret (token) and use it subsequently?
How are secrets different from globals?

Reply:
I don't think secrets should be mutable.